src/generators/mappersGenerator.ts

```diff
@@ -21,29 +28,132 @@ export function generateMappers(
     moduleSpecifier: "@azure/core-http"
   });
 
-  for (const mapper of clientDetails.mappers) {
-    // Generate the mapper definition and replace object key strings
-    // with the unquoted name
-    const mapperString = JSON.stringify(mapper, undefined, 2).replace(
-      /\"([^"]+)\":/g,
-      "$1:"
-    );
+  let discriminators: { [key: string]: string } = {};
 
+  for (const mapper of clientDetails.mappers) {
     if (mapper.type.name === MapperType.Composite && mapper.serializedName) {
+      const compositeMapper = mapper as CompositeMapper;
+
+      if (compositeMapper.type.polymorphicDiscriminator) {
+        const { uberParent, className } = compositeMapper.type;
+        if (!uberParent || !className) {
+          throw new Error(
+            `Expected CompositeMapper with polymorphicDiscriminator to specify uberParent property uberParent: ${uberParent} className: ${className}`
+          );
+        }
+
+        if (uberParent === className) {
+          discriminators[className] = className;
+        } else {
+          discriminators[`${uberParent}.${mapper.serializedName}`] = className;
+        }
+      }
+
       mappersFile.addVariableStatement({
         isExported: true,
         declarations: [
           {
-            name: normalizeName(mapper.serializedName, NameType.Class),
+            name: normalizeName(
+              compositeMapper.type.className || "",
+              NameType.Class
+            ),
             type: "coreHttp.CompositeMapper",
-            initializer: mapperString
+            initializer: writer => writeMapper(writer, mapper)
           }
         ],
         declarationKind: VariableDeclarationKind.Const,
         leadingTrivia: writer => writer.blankLine()
       });
     } else {
-      throw new Error("Mapper does not have a className");
+      throw new Error(
+        `Don't know how to create a mapper for type ${mapper.type.name}`
+      );
     }
   }
+
+  !isEmpty(discriminators) &&
+    mappersFile.addVariableStatement({
+      isExported: true,
+      declarations: [
+        {
+          name: "discriminators",
+          initializer: writer => {
+            writer.block(() => {
+              keys(discriminators).forEach(key => {
+                writer.write(`'${key}': ${discriminators[key]},`);
+              });
+            });
+          }
+        }
+      ]
+    });
 }
+
+const writeMapper = (writer: CodeBlockWriter, mapper: Mapper) => {
```

Comment:
@daviwil before, I was just doing a `JSON.stringify(mapper)` however since we need to spread the properties of the parent mapper and reference other mappers I moved to a WriterFunction. Do you know if there is a better/more elegant way to accomplish this?

i.e.

```typescript
export const Goblinshark: coreHttp.CompositeMapper = {
  serializedName: "goblin",
  type: {
    name: "Composite",
    className: "Goblinshark",
    uberParent: "Fish",
    polymorphicDiscriminator: Fish.type.polymorphicDiscriminator, // <- This
    modelProperties: {
      ...Shark.type.modelProperties, // <- This also
      jawsize: { type: { name: "Number" }, serializedName: "jawsize" },
      color: {
        type: { name: "String" },
        serializedName: "color",
        defaultValue: "gray"
      }
    }
  }
};
```